### ISSUE TITLE
Media & Text block: The link option does not work when the featured image is used

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#60497]
### Description

When the Media & Text block uses the featured image, and a link is added using the block toolbar option, the link does not work because it does not wrap around the image.

[It is not added to the block in index.php.](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/media-text/index.php#L32)

### Step-by-step reproduction instructions

Activate Gutenberg
Create a new post and add a featured image.
Add a Media & Text block and select the option "Use featured image".
In the block toolbar, select the link option and add a link.
Save and view the front.
Confirm that the image is not linked.

### Environment info

Gutenberg 18.0 or newer, or Gutenberg trunk.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by fabiankaegy** [comment]
This has also been brought up in the Core Editor channel in slack: https://wordpress.slack.com/archives/C02QB2JS7/p1716196185562119.

TLDR: We need help wit engineering
